cogito/put.go

```diff
@@ -277,14 +278,32 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeParseUrl parse the given URL and return an error if the URL is not well formated.
+// this avoid to eventually leak passwords in case of error
+//
+// From https://github.com/golang/go/issues/53993
+func safeParse(rawURL string) (*url.URL, error) {
```

Comment:
Due to the new URL parsing, in case of error, we only show the error and not the faulty URL. This means we can lose (important) debug information.
This is also causing tests `invalid HTTPS URL` and `invalid HTTP URL` to fail because they currently expect the URL in the response